translator/translate/otel/pipeline/prometheus/translator.go

```diff
@@ -12,41 +13,107 @@ import (
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/exporter/awsemf"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/exporter/prometheusremotewrite"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/extension/agenthealth"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/extension/sigv4auth"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/processor/batchprocessor"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/processor/rollupprocessor"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/receiver/adapter"
-)
-
-const (
-	pipelineName = "prometheus"
+	otelprom "github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/receiver/prometheus"
 )
 
 type translator struct {
+	name        string
+	dataType    component.DataType
+	destination string
```

Comment:
nit: There's a `common.DestinationProvider` (https://github.com/aws/amazon-cloudwatch-agent/blob/main/translator/translate/otel/common/options.go#L68) that can be embedded that will support the `common.WithDestination` option. Want to move away from defining the options in each of the translators. See https://github.com/aws/amazon-cloudwatch-agent/blob/main/translator/translate/otel/pipeline/jmx/translator.go#L50 for example usage.